In [ ]:
 import pandas as pd
import numpy as np
import plotly.express as px

In [ ]:
lang_df = pd.read_csv('Language enrollment.csv')

In [ ]:
lang_df.head(3)

INST_ID  SRVY_YEAR  TERM    YR&TERM            UNIV CAMPUS  NCES_ID  \
0   5210.0       1958  Fall  1958 Fall  A&M C OF TEXAS    NaN   228723   
1   5210.0       1958  Fall  1958 Fall  A&M C OF TEXAS    NaN   228723   
2   5210.0       1958  Fall  1958 Fall  A&M C OF TEXAS    NaN   228723   

                             UNIV_NAME_HISTORY             CITY  STATE_ID  \
0  A&M C of Texas renamed Texas A&M U in 1963.  College Station      44.0   
1  A&M C of Texas renamed Texas A&M U in 1963.  College Station      44.0   
2  A&M C of Texas renamed Texas A&M U in 1963.  College Station      44.0   

   ... UNDERGRAD_1_2  UNDERGRAD_3_4  UNDERGRAD_TOTAL GRAD_TOTAL  \
0  ...           NaN            NaN             86.0        NaN   
1  ...           NaN            NaN            148.0        NaN   
2  ...           NaN            NaN             66.0        NaN   

   ALL_LEVEL_TOTAL  LANG_REGION  OTHER_LANG  GEOGRAPHY_CODE  N_RESP ZERO_ERL  
0             86.0            E         NaN             4.0     NaN      NaN  
1            148.0            E         NaN             4.0     NaN      NaN  
2             66.0            E         NaN             4.0     NaN      NaN  

[3 rows x 24 columns]

In [ ]:
lang_df.columns

Index(['INST_ID', 'SRVY_YEAR', 'TERM', 'YR&TERM', 'UNIV', 'CAMPUS', 'NCES_ID',
       'UNIV_NAME_HISTORY', 'CITY', 'STATE_ID', 'STATE', 'MLA ICLEVEL',
       'LANG_CODE', 'LANGUAGE', 'UNDERGRAD_1_2', 'UNDERGRAD_3_4',
       'UNDERGRAD_TOTAL', 'GRAD_TOTAL', 'ALL_LEVEL_TOTAL', 'LANG_REGION',
       'OTHER_LANG', 'GEOGRAPHY_CODE', 'N_RESP', 'ZERO_ERL'],
      dtype='object')

In [ ]:
np.sum(lang_df['GRAD_TOTAL'])

552234.0

In [ ]:
lang_years_df = lang_df.groupby('SRVY_YEAR', as_index = False).agg({'ALL_LEVEL_TOTAL': 'sum'}).rename(columns={'SRVY_YEAR':'Year', 'ALL_LEVEL_TOTAL': 'Enrollment in languages'})
lang_years_df

Year  Enrollment in languages
0   1958                 433639.0
1   1959                 537927.0
2   1960                 612626.0
3   1961                 656607.0
4   1963                 821271.0
5   1965                1034651.0
6   1968                1124021.0
7   1969                 141901.0
8   1970                1153239.0
9   1971                 137615.0
10  1972                1007709.0
11  1974                 946310.0
12  1977                 934305.0
13  1980                 924337.0
14  1983                 965187.0
15  1986                1003173.0
16  1990                1185465.0
17  1995                1138772.0
18  1998                1186632.0
19  2002                1395807.0
20  2006                1575838.0
21  2009                1673566.0
22  2013                1561131.0
23  2016                1618859.0

-6.7% increase 2009-2013

3.7% increase 2013-2016

In [ ]:
lang_everything = lang_df.groupby(['SRVY_YEAR', 'LANGUAGE'], as_index = False).agg({'ALL_LEVEL_TOTAL': 'sum'}).rename(columns={'SRVY_YEAR':'Year', 'ALL_LEVEL_TOTAL': 'Enrollment in languages'})
lang_2009= lang_everything[lang_everything['Year'] == 2009]
lang_2016= lang_everything[lang_everything['Year'] == 2016]

In [ ]:
#All languages in 2009 with enrollment above 5000
lang_2009 = lang_2009[lang_2009['Enrollment in languages'] > 5000]
lang_2009 = lang_2009[['LANGUAGE', 'Enrollment in languages']]
lang_2009

LANGUAGE  Enrollment in languages
2214  AMERICAN SIGN LANGUAGE (ASL)                  92068.0
2218                        ARABIC                  34908.0
2255                       CHINESE                  59876.0
2282                        FRENCH                 215244.0
2287                        GERMAN                  95613.0
2293                GREEK, ANCIENT                  20033.0
2303              HEBREW, BIBLICAL                  13758.0
2305                HEBREW, MODERN                   8307.0
2323                       ITALIAN                  80322.0
2324                      JAPANESE                  72357.0
2333                        KOREAN                   8449.0
2337                         LATIN                  32446.0
2375                    PORTUGUESE                  11273.0
2381                       RUSSIAN                  26740.0
2402                       SPANISH                 861015.0

In [ ]:
#All languages in 2016 with enrollment above 5000
lang_2016 = lang_2016[lang_2016['Enrollment in languages'] > 5000]
lang_2016 = lang_2016[['LANGUAGE', 'Enrollment in languages']]
lang_2016

LANGUAGE  Enrollment in languages
2713  AMERICAN SIGN LANGUAGE (ASL)                 126030.0
2718                        ARABIC                  34171.0
2760                       CHINESE                  58102.0
2792                        FRENCH                 193134.0
2798                        GERMAN                  87798.0
2807                GREEK, ANCIENT                  11790.0
2818              HEBREW, BIBLICAL                  10391.0
2820                HEBREW, MODERN                   6005.0
2841                       ITALIAN                  61711.0
2842                      JAPANESE                  75229.0
2854                        KOREAN                  15072.0
2858                         LATIN                  26300.0
2907                    PORTUGUESE                  10596.0
2916                       RUSSIAN                  23044.0
2936                       SPANISH                 836280.0

In [ ]:
#Percent increase from 2009 to 2016 for languages with enrollment above 5000
compare_all = lang_2009.merge(lang_2016, on='LANGUAGE', how='inner').rename(columns={'Enrollment in languages_x': '2009', 'Enrollment in languages_y': '2016'})
compare_all['Percent increase'] = ((compare_all['2016'] - compare_all['2009'])/compare_all['2009']) * 100

In [ ]:
compare_all.sort_values(by = ['Percent increase'], ascending=False, inplace=True)
compare_all.reset_index(inplace = True)
compare_all = compare_all[['LANGUAGE', '2009', '2016', 'Percent increase']]

In [ ]:
compare_all

LANGUAGE      2009      2016  Percent increase
0                         KOREAN    8449.0   15072.0         78.387975
1   AMERICAN SIGN LANGUAGE (ASL)   92068.0  126030.0         36.887952
2                       JAPANESE   72357.0   75229.0          3.969208
3                         ARABIC   34908.0   34171.0         -2.111264
4                        SPANISH  861015.0  836280.0         -2.872772
5                        CHINESE   59876.0   58102.0         -2.962790
6                     PORTUGUESE   11273.0   10596.0         -6.005500
7                         GERMAN   95613.0   87798.0         -8.173575
8                         FRENCH  215244.0  193134.0        -10.272063
9                        RUSSIAN   26740.0   23044.0        -13.821990
10                         LATIN   32446.0   26300.0        -18.942242
11                       ITALIAN   80322.0   61711.0        -23.170489
12              HEBREW, BIBLICAL   13758.0   10391.0        -24.473034
13                HEBREW, MODERN    8307.0    6005.0        -27.711569
14                GREEK, ANCIENT   20033.0   11790.0        -41.147107

After Korean, ASL was the next fastest growing widely learned language from 2009 to 2016. ASL enrollment grew 37% within this time span.

In [ ]:
lang_df['LANGUAGE'].nunique()

432

In [ ]:
fig = px.scatter(lang_years_df, x="Year", y="Enrollment in languages", title='College Student Enrollment in Language Courses')
fig.show()

In [ ]:
lang_df[lang_df['LANGUAGE'] == 'KOREAN']['LANG_CODE'].unique()
#Korean LANG_CODE is 47.0

array([47.])

In [ ]:
np.sum(lang_df['LANG_CODE'] == 47.0)
##check if total matches below cell

1192

In [ ]:
np.sum(lang_df['LANGUAGE'] == 'KOREAN')
##1192 entries of Korean

1192

In [ ]:
kor_df = lang_df[lang_df['LANG_CODE'] == 47.0]

In [ ]:
kor_years_df = kor_df.groupby('SRVY_YEAR', as_index = False).agg({'ALL_LEVEL_TOTAL': 'sum'}).rename(columns={'SRVY_YEAR':'Year', 'ALL_LEVEL_TOTAL': 'Enrollment in Korean'})
kor_years_df = kor_years_df.merge(lang_years_df, on='Year', how='right')

In [ ]:
kor_years_df['Enrollment in Korean'] = kor_years_df['Enrollment in Korean'].fillna(0)

In [ ]:
kor_years_df['Korean percent'] = (kor_years_df['Enrollment in Korean']/kor_years_df['Enrollment in languages'])*100

In [ ]:
kor_years_df

Year  Enrollment in Korean  Enrollment in languages  Korean percent
0   1958                  26.0                 433639.0        0.005996
1   1959                  25.0                 537927.0        0.004647
2   1960                 168.0                 612626.0        0.027423
3   1961                 190.0                 656607.0        0.028937
4   1963                 182.0                 821271.0        0.022161
5   1965                  82.0                1034651.0        0.007925
6   1968                  70.0                1124021.0        0.006228
7   1969                  13.0                 141901.0        0.009161
8   1970                 101.0                1153239.0        0.008758
9   1971                  25.0                 137615.0        0.018167
10  1972                   0.0                1007709.0        0.000000
11  1974                  87.0                 946310.0        0.009194
12  1977                 163.0                 934305.0        0.017446
13  1980                 365.0                 924337.0        0.039488
14  1983                 660.0                 965187.0        0.068381
15  1986                 875.0                1003173.0        0.087223
16  1990                2375.0                1185465.0        0.200343
17  1995                3343.0                1138772.0        0.293562
18  1998                4479.0                1186632.0        0.377455
19  2002                5211.0                1395807.0        0.373332
20  2006                7146.0                1575838.0        0.453473
21  2009                8449.0                1673566.0        0.504850
22  2013               12256.0                1561131.0        0.785072
23  2016               15072.0                1618859.0        0.931026

45% increase 2009-2013

23% increase 2013-2016

**Slight** discrepency in 2013: NPR reported in 2015 that the year 2013 had enrollment at 12,229 rather than 12,256. Not sure how...

In [ ]:
fig = px.scatter(kor_years_df, x="Year", y="Enrollment in Korean", title='College Student Enrollment in Korean Language')
fig.show()

In [ ]:
fig = px.scatter(kor_years_df, x="Year", y="Korean percent", title='Percent of students studying a language enrolled in Korean')
fig.show()

In [ ]:
kor_colleges_df = kor_df.groupby('SRVY_YEAR', as_index = False)['UNIV'].nunique().rename(columns={'SRVY_YEAR':'Year', 'UNIV': 'Number of colleges offering Korean'})
kor_colleges_df = kor_colleges_df.merge(lang_years_df, on='Year', how='right')[['Year', 'Number of colleges offering Korean']]
kor_colleges_df['Number of colleges offering Korean'] = kor_colleges_df['Number of colleges offering Korean'].fillna(0)

In [ ]:
kor_colleges_df

Year  Number of colleges offering Korean
0   1958                                 1.0
1   1959                                 1.0
2   1960                                 7.0
3   1961                                 1.0
4   1963                                 1.0
5   1965                                 9.0
6   1968                                 7.0
7   1969                                 1.0
8   1970                                 1.0
9   1971                                 1.0
10  1972                                 0.0
11  1974                                 8.0
12  1977                                10.0
13  1980                                16.0
14  1983                                23.0
15  1986                                30.0
16  1990                                50.0
17  1995                                72.0
18  1998                                77.0
19  2002                               102.0
20  2006                               125.0
21  2009                               147.0
22  2013                               162.0
23  2016                               171.0

In [ ]:
fig = px.scatter(kor_colleges_df, x="Year", y="Number of colleges offering Korean", title='Number of colleges offering Korean Language')
fig.show()

In [ ]:
lang_df[lang_df['LANGUAGE'] == 'CHINESE']['LANG_CODE'].unique()
##Chinese LANG_CODE is 43.0

array([43.])

In [ ]:
chin_df = lang_df[lang_df['LANG_CODE'] == 43.0]
len(chin_df)
#7337 entries of Chinese

7337

In [ ]:
chin_years_df = chin_df.groupby('SRVY_YEAR', as_index = False).agg({'ALL_LEVEL_TOTAL': 'sum'}).rename(columns={'SRVY_YEAR':'Year', 'ALL_LEVEL_TOTAL': 'Enrollment in Chinese'})

In [ ]:
chin_years_df

Year  Enrollment in Chinese
0   1958                  615.0
1   1959                  959.0
2   1960                  679.0
3   1961                 2200.0
4   1963                 2444.0
5   1965                 3359.0
6   1968                 5061.0
7   1969                  685.0
8   1970                 6115.0
9   1971                  953.0
10  1974                10576.0
11  1977                 9798.0
12  1980                11366.0
13  1983                13178.0
14  1986                16891.0
15  1990                19427.0
16  1995                26471.0
17  1998                28456.0
18  2002                34153.0
19  2006                51382.0
20  2009                59876.0
21  2013                61084.0
22  2016                58102.0

2% increase 2009-2013

-4.9% increase 2013-2016

In [ ]:
fig = px.scatter(chin_years_df, x="Year", y="Enrollment in Chinese", title='College Student Enrollment in Chinese Language')
fig.show()

In [ ]:
chin_colleges_df = chin_df.groupby('SRVY_YEAR', as_index = False)['UNIV'].nunique().rename(columns={'SRVY_YEAR':'Year', 'UNIV': 'Number of colleges offering Chinese'})
chin_colleges_df

Year  Number of colleges offering Chinese
0   1958                                    1
1   1959                                    2
2   1960                                   21
3   1961                                    1
4   1963                                    1
5   1965                                   80
6   1968                                  111
7   1969                                   15
8   1970                                    1
9   1971                                    1
10  1974                                  180
11  1977                                  203
12  1980                                  219
13  1983                                  243
14  1986                                  331
15  1990                                  412
16  1995                                  393
17  1998                                  422
18  2002                                  543
19  2006                                  659
20  2009                                  819
21  2013                                  923
22  2016                                  881

In [ ]:
fig = px.scatter(chin_colleges_df, x="Year", y="Number of colleges offering Chinese", title='Number of colleges offering Chinese')
fig.show()

In [ ]:
lang_df[lang_df['LANGUAGE'] == 'JAPANESE']['LANG_CODE'].unique()
##Japanese LANG_CODE is 42.0

array([42.])

In [ ]:
jap_df = lang_df[lang_df['LANG_CODE'] == 42.0]
len(jap_df)
#7857 entries of Japanese

7857

In [ ]:
jap_years_df = jap_df.groupby('SRVY_YEAR', as_index = False).agg({'ALL_LEVEL_TOTAL': 'sum'}).rename(columns={'SRVY_YEAR':'Year', 'ALL_LEVEL_TOTAL': 'Enrollment in Japanese'})
jap_years_df

Year  Enrollment in Japanese
0   1958                   844.0
1   1959                  1271.0
2   1960                  1754.0
3   1961                  1976.0
4   1963                  2718.0
5   1965                  3505.0
6   1968                  4322.0
7   1969                   568.0
8   1970                  6620.0
9   1971                   760.0
10  1974                  9604.0
11  1977                 10721.0
12  1980                 11516.0
13  1983                 16126.0
14  1986                 23454.0
15  1990                 45830.0
16  1995                 44723.0
17  1998                 43141.0
18  2002                 52238.0
19  2006                 65410.0
20  2009                 72357.0
21  2013                 66771.0
22  2016                 75229.0

-7.7% increase 2009-2013

12.7% increase 2013-2016

In [ ]:
fig = px.scatter(jap_years_df, x="Year", y="Enrollment in Japanese", title='College Student Enrollment in Japanese Language')
fig.show()

In [ ]:
jap_colleges_df = jap_df.groupby('SRVY_YEAR', as_index = False)['UNIV'].nunique().rename(columns={'SRVY_YEAR':'Year', 'UNIV': 'Number of colleges offering Japanese'})
jap_colleges_df

Year  Number of colleges offering Japanese
0   1958                                     1
1   1959                                     2
2   1960                                    32
3   1961                                     1
4   1963                                     1
5   1965                                    59
6   1968                                    75
7   1969                                    18
8   1970                                     1
9   1971                                     1
10  1974                                   166
11  1977                                   196
12  1980                                   214
13  1983                                   274
14  1986                                   368
15  1990                                   657
16  1995                                   698
17  1998                                   684
18  2002                                   782
19  2006                                   726
20  2009                                   741
21  2013                                   744
22  2016                                   711

In [ ]:
fig = px.scatter(jap_colleges_df, x="Year", y="Number of colleges offering Japanese", title='Number of colleges offering Japanese')
fig.show()